packages/pyright-internal/src/languageService/analyzerServiceExecutor.ts

```diff
@@ -120,33 +118,25 @@ function getEffectiveCommandLineOptions(
     }
 
     if (serverSettings.venvPath) {
-        commandLineOptions.venvPath = combinePaths(
-            workspaceRootPath || languageServiceRootPath,
-            serverSettings.venvPath
-        );
+        commandLineOptions.venvPath = serverSettings.venvPath.getFilePath();
     }
 
-    if (serverSettings.pythonPath) {
+    if (serverSettings.pythonPath && serverSettings.pythonPath.getRootPathLength() > 0) {
         // The Python VS Code extension treats the value "python" specially. This means
         // the local python interpreter should be used rather than interpreting the
         // setting value as a path to the interpreter. We'll simply ignore it in this case.
-        if (!isPythonBinary(serverSettings.pythonPath)) {
-            commandLineOptions.pythonPath = combinePaths(
-                workspaceRootPath || languageServiceRootPath,
-                serverSettings.pythonPath
-            );
-        }
+        commandLineOptions.pythonPath = serverSettings.pythonPath.getFilePath();
```

Comment:
It was a side effect of the pythonPath in the settings becoming a URI. It will never refer to just `python` or `python3` anymore. 

The code that sets the settings.pythonPath checks for this case already, so the old check of setting it when it wasn't `python` or `python3` isn't necessary (or really possible anymore). 

To be honest, I'm not 100% sure this is the correct thing to do though. Like if the pythonPath was set to a relative path, it would come in here as a rooted path, so it seems like it should work, but that means there's a dependency between that happening and this code. If it wasn't rooted, it seems like this should still use the file path though. I'll change just that.